### DISCUSSION TITLE
MissingSpatialDimensionError when merging tiles from Polaris

### DISCUSSION THREAD

**Discussion by lucap1211** [earthlab-education/Earth-Analytics-2023-01-Intro#682] [! GitHub help]
As I try to download more than one tile to cover my grassland's (Fort Pierre National Grassland) total area, I run into this "MissingSpatialDimensionError: MissingSpatialDimensionError: x dimension not found. 'rio.set_spatial_dims()' or using 'rename()' to change the dimension name to 'x' can address this" error. Additionally, I noticed that plotting the polaris ph mean data with hvplot(rasterize=True) has an output map where the lat/lon dimensions are switched.

I tried this example code and also  .rename_dims() instead of the .set_spatial_dims()

<code>
min_lat, min_lon = 43, -101
max_lat, max_lon = min_lat+1, min_lon+1
polaris_url_format = (
    "http://hydrology.cee.duke.edu/POLARIS/PROPERTIES/v1.0"
    "/ph/mean/60_100/lat{min_lat}{max_lat}_lon{min_lon}{max_lon}.tif"
)
polaris_url_fp_1 = polaris_url_format.format(
    min_lat=min_lat, min_lon=min_lon, max_lat=max_lat, max_lon=max_lon)
ph_fp_da_1 = rxr.open_rasterio(polaris_url_fp_1, masked=True).squeeze()
ph_fp_da_1.rio.set_spatial_dims('x', 'y', inplace=True)
min_lat, min_lon = 44, -101
max_lat, max_lon = min_lat+1, min_lon+1
polaris_url_format = (
    "http://hydrology.cee.duke.edu/POLARIS/PROPERTIES/v1.0"
    "/ph/mean/60_100/lat{min_lat}{max_lat}_lon{min_lon}{max_lon}.tif"
)
polaris_url_fp_2 = polaris_url_format.format(
    min_lat=min_lat, min_lon=min_lon, max_lat=max_lat, max_lon=max_lon)
ph_fp_da_2 = rxr.open_rasterio(polaris_url_fp_2, masked=True).squeeze()
ph_fp_da_2.rio.set_spatial_dims('x', 'y', inplace=True)
ph_fp_da = rxrmerge.merge_arrays(ph_fp_da_1, ph_fp_da_2)
</code>

I am running my code on a local computer with the earth-analytics environment and jupyter lab.  

Thanks for any help!

**Comment by ellenalamont17** [comment]
It looks like you are missing the reprojection part. See the sample code below. Also, make sure that your grassland unit boundary is also reprojected to the same UTM EPSG or the clipping will be weird. I could not run this code to test it with your variables/boundaries, but mine looks similar to this. Hope it helps!

``````
# Format URL for FP-1
min_lat, min_lon = 43, -101
max_lat, max_lon = min_lat+1, min_lon+1
polaris_url_format = (
    "http://hydrology.cee.duke.edu/POLARIS/PROPERTIES/v1.0"
    "/ph/mean/60_100/lat{min_lat}{max_lat}_lon{min_lon}{max_lon}.tif"
)
polaris_url_fp_1 = polaris_url_format.format(
    min_lat=min_lat, min_lon=min_lon, max_lat=max_lat, max_lon=max_lon)


# Format URL for FP-2
min_lat, min_lon = 44, -101
max_lat, max_lon = min_lat+1, min_lon+1
polaris_url_format = (
    "http://hydrology.cee.duke.edu/POLARIS/PROPERTIES/v1.0"
    "/ph/mean/60_100/lat{min_lat}{max_lat}_lon{min_lon}{max_lon}.tif"
)
polaris_url_fp_2 = polaris_url_format.format(
    min_lat=min_lat, min_lon=min_lon, max_lat=max_lat, max_lon=max_lon)


# Initiate an empty list or dictionary to hold your URLs
polaris_das = []

# Open your URLs and append to list
polaris_das.append(
        rxr.open_rasterio(polaris_url,_fp_1 masked=True).squeeze())
polaris_das.append(
        rxr.open_rasterio(polaris_url,_fp_2 masked=True).squeeze())

# Merge, reproject, and clip
ph_da = (
    rxrmerge.merge_arrays(polaris_das)
    .rio.reproject(00000)                               # Replace 00000 with your region's UTM EPSG 
    .rio.clip_box(*grass_da.total_bounds))    # Replace grass_da with your reprojected unit boundary variable
``````
